diff --git docutils/writers/html4css1/__init__.py docutils/writers/html4css1/__init__.py
index 39b0a27..ef2af0e 100644
--- docutils/writers/html4css1/__init__.py
+++ docutils/writers/html4css1/__init__.py
@@ -327,6 +327,7 @@ class HTMLTranslator(nodes.NodeVisitor):
         self.in_mailto = False
         self.author_in_authors = False
         self.math_header = []
+        self.settings.file_insertion_enabled = True
 
     def astext(self):
         return ''.join(self.head_prefix + self.head
