diff --git Orange/canvas/__main__.py Orange/canvas/__main__.py
index adc7417..ca538e7 100644
--- Orange/canvas/__main__.py
+++ Orange/canvas/__main__.py
@@ -228,7 +228,7 @@ def main(argv=None):
             pkg_name = canvas.__name__
             resource = "styles/" + stylesheet
 
-            if pkg_resources.resource_exists(pkg_name, resource):
+            if 0:
                 stylesheet_string = \
                     pkg_resources.resource_string(pkg_name, resource).decode()
 
