--- configure-ng.old.py	2016-05-11 22:08:45 -0500
+++ configure-ng.py	2016-05-11 22:17:00 -0500
@@ -462,7 +462,7 @@
         self.no_pydbus = False
         self.no_tools = False
         self.prot_is_public = (self.py_platform.startswith('linux') or self.py_platform == 'darwin')
-        self.qmake = self._find_exe('qmake')
+        self.qmake = self._find_exe('qmake-qt4')
         self.qmake_variables = []
         self.py_pylib_dir = ''
         self.py_pylib_lib = ''
