--- setup.py    2013-11-10 08:36:41.000000000 +0100
+++ setup.py    2014-03-31 01:56:36.101247755 +0200
@@ -35,12 +35,16 @@
 # This global variable is used to hold the list of modules to be disabled.
 disabled_module_list = []
 
-def add_dir_to_list(dirlist, dir):
-    """Add the directory 'dir' to the list 'dirlist' (at the front) if
+def add_dir_to_list(dirlist, dir, append=False):
+    """Add the directory 'dir' to the list 'dirlist' (at the front if append
+    is False; at the back if append is True) if
     1) 'dir' is not already in 'dirlist'
     2) 'dir' actually exists, and is a directory."""
     if dir is not None and os.path.isdir(dir) and dir not in dirlist:
-        dirlist.insert(0, dir)
+        if append:
+            dirlist.append(dir)
+        else:
+            dirlist.insert(0, dir)
 
 def macosx_sdk_root():
     """
@@ -508,6 +512,6 @@
                 '/lib64', '/usr/lib64',
                 '/lib', '/usr/lib',
                 ):
-                add_dir_to_list(lib_dirs, d)
+                add_dir_to_list(lib_dirs, d, append=True)
         exts = []
         missing = []
