diff --git setup.py setup.py
index 001f0f3..1c99fe7 100644
--- setup.py
+++ setup.py
@@ -139,7 +139,8 @@ setup(name='zope.interface',
         "Topic :: Software Development :: Libraries :: Python Modules",
       ],
 
-      packages = ['zope', 'zope.interface', 'zope.interface.tests'],
+      packages = ['zope', 'zope.interface', 'zope.interface.tests',
+                  'zope.interface.common', 'zope.interface.common.tests'],
       package_dir = {'': 'src'},
       cmdclass = {'build_ext': optional_build_ext,
                   },
