diff --git setup_build.py setup_build.py
index ffe19b7..f3c97d7 100644
--- setup_build.py
+++ setup_build.py
@@ -39,7 +39,10 @@ COMPILER_SETTINGS = {
    'libraries'      : ['hdf5', 'hdf5_hl'],
    'include_dirs'   : [localpath('lzf')],
    'library_dirs'   : [],
-   'define_macros'  : [('H5_USE_16_API', None)]
+   'define_macros'  : [('H5_USE_16_API', None),
+                       ('_HDF5USEDLL_', None),
+                       ('H5_BUILT_AS_DYNAMIC_LIB', None),
+                      ]
 }
 
 if sys.platform.startswith('win'):
