diff --git Modules/readline.c Modules/readline.c
index 06ae7c6..3633c34 100644
--- Modules/readline.c
+++ Modules/readline.c
@@ -615,8 +615,6 @@ PyDoc_STRVAR(doc_get_line_buffer,
 return the current contents of the line buffer.");
 
 
-#ifdef HAVE_RL_COMPLETION_APPEND_CHARACTER
-
 /* Exported function to clear the current history */
 
 static PyObject *
@@ -629,7 +629,6 @@ py_clear_history(PyObject *self, PyObject *noarg)
 PyDoc_STRVAR(doc_clear_history,
 "clear_history() -> None\n\
 Clear the current readline history.");
-#endif
 
 
 /* Exported function to insert text into the line buffer */
@@ -708,9 +708,7 @@ static struct PyMethodDef readline_methods[] =
     {"set_pre_input_hook", set_pre_input_hook,
      METH_VARARGS, doc_set_pre_input_hook},
 #endif
-#ifdef HAVE_RL_COMPLETION_APPEND_CHARACTER
     {"clear_history", py_clear_history, METH_NOARGS, doc_clear_history},
-#endif
     {0, 0}
 };
 
