diff --git Lib/site.py Lib/site.py
index 3b51e81..fc19620 100644
--- Lib/site.py
+++ Lib/site.py
@@ -535,6 +535,9 @@ def main():
     # this module is run as a script, because this code is executed twice.
     if hasattr(sys, "setdefaultencoding"):
         del sys.setdefaultencoding
+    if sys.platform == 'win32':
+        os.environ["PATH"] = r"%s\Library\bin;%s" % (sys.prefix,
+                                                     os.environ["PATH"])
 
 main()
 
