diff --git setup.py setup.py
index 51e8d02..ff6451e 100644
--- setup.py
+++ setup.py
@@ -21,12 +21,6 @@ setup(
             'src/lz4hc.c',
             'src/python-lz4.c'
         ], extra_compile_args=[
-            "-std=c99",
-            "-O3",
-            "-Wall",
-            "-W",
-            "-Wundef",
-            "-DLZ4_VERSION=\"r119\"",
         ])
     ],
     setup_requires=["nose>=1.0"],
