diff --git setup.py setup.py
index 74b8e52..4ad99d5 100644
--- setup.py
+++ setup.py
@@ -39,15 +39,15 @@ setup(
     download_url = 'http://pypi.python.org/pypi/smart_open',
 
     keywords = 'file streaming, s3, hdfs',
-
+    zip_safe = False,
     license = 'MIT',
     platforms = 'any',
 
     install_requires=[
         'boto >= 2.32',
-        'httpretty==0.8.10',
+        'httpretty',
         'bz2file',
-        'requests==2.8.1',
+        'requests',
     ],
 
     test_suite="smart_open.tests",
