diff --git setup.py setup.py
index 711638f..a42933d 100755
--- setup.py
+++ setup.py
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 import os
-from setuptools import setup, find_packages
+from distutils.core import setup
 
 version = __import__('unicodecsv').__version__
 
@@ -12,7 +12,7 @@ setup(
     author='Jeremy Dunck',
     author_email='jdunck@gmail.com',
     url='https://github.com/jdunck/python-unicodecsv',
-    packages=find_packages(),
+    packages=['unicodecsv'],
     tests_require=['unittest2>=0.5.1'],
     test_suite='runtests.get_suite',
     license='BSD License',
