diff --git pip/__init__.py pip/__init__.py
index a6caf04..ff057e6 100755
--- pip/__init__.py
+++ pip/__init__.py
@@ -197,6 +197,19 @@ def main(args=None):
     if args is None:
         args = sys.argv[1:]
 
+    if 'virtualenv' in args:
+        print("""
+WARNING: using virtualenv with conda is untested and not recommended.
+    We suggest using the conda command to create environments instead.
+    For more information about creating conda environments, please see:
+        http://docs.continuum.io/conda/examples/create.html
+""")
+        sys.stdout.write("Proceed (y/n)? ")
+        sys.stdout.flush()
+        proceed = sys.stdin.readline()
+        if proceed.strip().lower() not in ('y', 'yes'):
+            sys.exit(0)
+
     # Enable our Deprecation Warnings
     for deprecation_warning in deprecation.DEPRECATIONS:
         warnings.simplefilter("default", deprecation_warning)
