diff --git astropy/utils/compat/numpycompat.py astropy/utils/compat/numpycompat.py
index fbdd029..fc7aadd 100644
--- astropy/utils/compat/numpycompat.py
+++ astropy/utils/compat/numpycompat.py
@@ -21,7 +21,7 @@ NUMPY_LT_1_9_1 = not minversion('numpy', '1.9.1')
 NUMPY_LT_1_10 = not minversion('numpy', '1.10.0')
 NUMPY_LT_1_10_4 = not minversion('numpy', '1.10.4')
 NUMPY_LT_1_11 = not minversion('numpy', '1.11.0')
-NUMPY_LT_1_12 = not minversion('numpy', '1.12dev')
+NUMPY_LT_1_12 = not minversion('numpy', '1.12.0')
 
 
 def _monkeypatch_unicode_mask_fill_values():
