diff --git setup.py setup.py
index 39a69ad..4f113bc 100644
--- setup.py
+++ setup.py
@@ -6,9 +6,9 @@ from setuptools import setup
 install_requires = [
     'xlrd>=0.7.1',
     'sqlalchemy>=0.6.6',
-    'openpyxl==2.2.0-b1',
+    'openpyxl',
     'six>=1.6.1',
-    'python-dateutil==2.2'
+    'python-dateutil'
 ]
 
 if sys.version_info < (2, 7):
@@ -17,7 +17,7 @@ if sys.version_info < (2, 7):
     install_requires.append('simplejson>=3.6.3')
 
 if sys.version_info[0] == 2:
-    install_requires.append('dbf==0.94.003')
+    install_requires.append('dbf')
 
 setup(
     name='csvkit',
