diff --git lib/matplotlib/__init__.py lib/matplotlib/__init__.py
index 5c749ba..af52e5b 100644
--- lib/matplotlib/__init__.py
+++ lib/matplotlib/__init__.py
@@ -1,4 +1,4 @@
-"""
+﻿"""
 This is an object-oriented plotting library.
 
 A procedural interface is provided by the companion pyplot module,
@@ -1486,7 +1486,6 @@ default_test_modules = [
     'matplotlib.tests.test_basic',
     'matplotlib.tests.test_bbox_tight',
     'matplotlib.tests.test_cbook',
-    'matplotlib.tests.test_coding_standards',
     'matplotlib.tests.test_collections',
     'matplotlib.tests.test_colorbar',
     'matplotlib.tests.test_colors',
