diff --git matplotlibrc.template matplotlibrc.template
index f5dce98..5d3c2b4 100644
--- matplotlibrc.template
+++ matplotlibrc.template
@@ -35,7 +35,7 @@
 # You can also deploy your own backend outside of matplotlib by
 # referring to the module name (which must be in the PYTHONPATH) as
 # 'module://my_backend'.
-backend      : $TEMPLATE_BACKEND
+backend      : Qt5Agg
 
 # If you are using the Qt4Agg backend, you can choose here
 # to use the PyQt4 bindings or the newer PySide bindings to
