diff --git Lib/distutils/cygwinccompiler.py Lib/distutils/cygwinccompiler.py
index d28b1b3..a03e04f 100644
--- Lib/distutils/cygwinccompiler.py
+++ Lib/distutils/cygwinccompiler.py
@@ -137,11 +137,11 @@ class CygwinCCompiler(UnixCCompiler):
 
         # Hard-code GCC because that's what this is all about.
         # XXX optimization, warnings etc. should be customizable.
-        self.set_executables(compiler='gcc -mcygwin -O -Wall',
-                             compiler_so='gcc -mcygwin -mdll -O -Wall',
-                             compiler_cxx='g++ -mcygwin -O -Wall',
-                             linker_exe='gcc -mcygwin',
-                             linker_so=('%s -mcygwin %s' %
+        self.set_executables(compiler='gcc @DMSW@ -Wall',
+                             compiler_so='gcc @DMSW@ -mdll -O -Wall',
+                             compiler_cxx='g++ @DMSW@ -O -Wall',
+                             linker_exe='gcc',
+                             linker_so=('%s %s' %
                                         (self.linker_dll, shared_option)))
 
         # cygwin and mingw32 need different sets of libraries
@@ -401,5 +401,5 @@ def get_versions():
 
 def is_cygwingcc():
     '''Try to determine if the gcc that would be used is from cygwin.'''
-    out_string = check_output(['gcc', '-dumpmachine'])
+    out_string = check_output([find_executable('gcc'), '-dumpmachine'])
     return out_string.strip().endswith(b'cygwin')
