diff --git Lib/site.py Lib/site.py
index afa8217..7ce355a 100644
--- Lib/site.py
+++ Lib/site.py
@@ -557,6 +557,9 @@ def main():
     execsitecustomize()
     if ENABLE_USER_SITE:
         execusercustomize()
+    if sys.platform == 'win32':
+        os.environ["PATH"] = r"%s\Library\bin;%s" % (sys.prefix,
+                                                     os.environ["PATH"])
     try:
         import anaconda_decrypt
     except ImportError:
