--- src\algorithm\LineIntersector.cpp.orig	2013-08-26 01:10:29.000000000 +1000
+++ src\algorithm\LineIntersector.cpp	2015-11-13 11:49:38.744850900 +1000
@@ -17,6 +17,11 @@
  *
  **********************************************************************/
 
+#if _MSC_VER >= 1900
+#include <algorithm>
+#define max(x, y) std::max(x, y)
+#define min(x, y) std::min(x, y)
+#endif
 #include <geos/algorithm/LineIntersector.h>
 #include <geos/algorithm/CGAlgorithms.h>
 #include <geos/algorithm/HCoordinate.h>
