--- src\operation\buffer\OffsetCurvesetBuilder.cpp.orig	2013-08-26 01:10:29.000000000 +1000
+++ src\operation\buffer\OffsetCurvesetBuilder.cpp	2015-11-13 12:21:59.687233700 +1000
@@ -17,7 +17,12 @@
  * Last port: operation/buffer/OffsetCurveSetBuilder.java r378 (JTS-1.12)
  *
  **********************************************************************/
-
+#if _MSC_VER >= 1900
+#include <algorithm>
+#define max(x, y) std::max(x, y)
+#define min(x, y) std::min(x, y)
+#define abs(x) std::abs(x)
+#endif
 #include <geos/algorithm/CGAlgorithms.h>
 #include <geos/algorithm/MinimumDiameter.h>
 #include <geos/util/UnsupportedOperationException.h>
@@ -333,8 +338,8 @@
 		return isTriangleErodedCompletely(ringCoord, bufferDistance);
 
   const Envelope* env = ring->getEnvelopeInternal();
-  double envMinDimension = std::min(env->getHeight(), env->getWidth());
-  if (bufferDistance < 0.0 && 2 * std::abs(bufferDistance) > envMinDimension)
+  double envMinDimension = min(env->getHeight(), env->getWidth());
+  if (bufferDistance < 0.0 && 2 * abs(bufferDistance) > envMinDimension)
       return true;
 
 	/**
