--- setup.py
+++ setup.py
@@ -39,9 +39,9 @@ Links
 
 """
 from __future__ import print_function
-from setuptools import Command, setup
+from distutils.core import setup
 
-class run_audit(Command):
+class run_audit(object):
     """Audits source code using PyFlakes for following issues:
         - Names which are used but not defined or used before they are defined.
         - Names which are redefined without having been used.
@@ -107,6 +107,6 @@ setup(
         'Topic :: Internet :: WWW/HTTP :: Dynamic Content',
         'Topic :: Software Development :: Libraries :: Python Modules'
     ],
-    cmdclass={'audit': run_audit},
+    #cmdclass={'audit': run_audit},
     test_suite='flask.testsuite.suite'
 )
