diff --git Lib/site.py Lib/site.py
index 0fc9200..f20c692 100644
--- Lib/site.py
+++ Lib/site.py
@@ -537,6 +537,9 @@ def main():
     execsitecustomize()
     if ENABLE_USER_SITE:
         execusercustomize()
+    if sys.platform == 'win32':
+        os.environ["PATH"] = r"%s\Library\bin;%s" % (sys.prefix,
+                                                     os.environ["PATH"])
 
 # Prevent extending of sys.path when python was started with -S and
 # site is imported later.
