diff --git Lib/distutils/spawn.py Lib/distutils/spawn.py
index e5654ff..2a29c92 100644
--- Lib/distutils/spawn.py
+++ Lib/distutils/spawn.py
@@ -184,18 +184,19 @@ def find_executable(executable, path=None):
     """
     if path is None:
         path = os.environ['PATH']
-    paths = string.split(path, os.pathsep)
-    (base, ext) = os.path.splitext(executable)
-    if (sys.platform == 'win32' or os.name == 'os2') and (ext != '.exe'):
-        executable = executable + '.exe'
-    if not os.path.isfile(executable):
-        for p in paths:
-            f = os.path.join(p, executable)
-            if os.path.isfile(f):
-                # the file exists, we have a shot at spawn working
-                return f
-        return None
-    else:
-        return executable
+    paths = path.split(os.pathsep)
+
+    for ext in '.exe', '.bat', '':
+        newexe = executable + ext
+
+        if os.path.isfile(newexe):
+            return newexe
+        else:
+            for p in paths:
+                f = os.path.join(p, newexe)
+                if os.path.isfile(f):
+                    # the file exists, we have a shot at spawn working
+                    return f
+    return None
 
 # find_executable()
