diff --git numpy/__init__.py numpy/__init__.py
index d4ef54d..9318d75 100644
--- numpy/__init__.py
+++ numpy/__init__.py
@@ -108,6 +108,10 @@ from __future__ import division, absolute_import, print_function
 
 import sys
 
+import ctypes
+_old_rtld = sys.getdlopenflags()
+sys.setdlopenflags(_old_rtld | ctypes.RTLD_GLOBAL)
+
 
 class ModuleDeprecationWarning(DeprecationWarning):
     """Module deprecation warning.
@@ -227,3 +231,9 @@ else:
     warnings.filterwarnings("ignore", message="numpy.dtype size changed")
     warnings.filterwarnings("ignore", message="numpy.ufunc size changed")
     warnings.filterwarnings("ignore", message="numpy.ndarray size changed")
+
+sys.setdlopenflags(_old_rtld)
+del _old_rtld
+del sys
+del ctypes
+__mkl_version__ = "__MKL_VERSION__"
