diff --git setup-proj.py setup-proj.py
index 6d2f797..893ce65 100644
--- setup-proj.py
+++ setup-proj.py
@@ -9,17 +9,16 @@ proj_libdir = os.environ.get('PROJ_LIBDIR')
 proj_incdir = os.environ.get('PROJ_INCDIR')
 libdirs=[]
 incdirs=[numpy.get_include()]
-libraries=['proj']
+libraries=['proj_i']
 
 if proj_libdir is None and proj_dir is not None:
     libdirs.append(os.path.join(proj_dir,'lib'))
-    libdirs.append(os.path.join(proj_dir,'lib64'))
 if proj_incdir is None and proj_dir is not None:
     incdirs.append(os.path.join(proj_dir,'include'))
 
 pyprojext =\
 Extension("pyproj._proj",["_proj.c"],include_dirs=incdirs,library_dirs=libdirs,\
-runtime_library_dirs=libdirs,libraries=libraries)
+libraries=libraries)
 
 # over-write default data directory.
 pyproj_datadir = os.path.join(os.path.join(proj_dir,'share'),'proj')
